### PR TITLE
Add dashboard_client definition

### PR DESCRIPTION
**Pull request by junxu**
- Add DashboardClient struct according osbapi spec v2.12
 - Add doumentation link to the v2.12 tag description of the dashboard_client

### CODE REVIEW

File: pkg/brokerapi/service.go [hunk 1]
```diff
@@ -21,14 +21,24 @@ package brokerapi
...
+}
+
+// DashboardClient represents a dashboard_client which contains the data necessary to activate
+// the Dashboard SSO feature for this service.
+//
+// https://github.com/openservicebrokerapi/servicebroker/blob/v2.12/spec.md#dashboard-client-object
+type DashboardClient struct {
+	ID          string `json:"id,omitempty"`
+	Secret      string `json:"secret,omitempty"`
+	RedirectURI string `json:"redirect_uri,omitempty"`
```
Comment (duglin) [inline pkg/brokerapi/service.go]: The downsize to this is that we lose any extension properties that might have been in there.  The spec doesn't say whether there could be any or not though - we should get that fixed - but my preference would be to support them.  Is there a way to not lose them but still have these 3 be well defined?

Comment (junxu) [reply pkg/brokerapi/service.go]: what about add one more field for extension ?
   Config interface{} `json: "config,omitempty"`

Comment (duglin) [reply pkg/brokerapi/service.go]: for now we can just keep this as-is and wait for the spec to decide what they want to do.

**Comment by pmorie** [comment]
I don't understand why this PR is needed.  We should move this package into contrib/, actually - something I have been planning on creating an issue for.  As far as I know, this package is not used anywhere except in the ups broker in contrib.

**Comment by duglin** [comment]
@pmorie can you elaborate on which package you're referring to?

**Comment by pmorie** [comment]
@duglin - `pkg/brokerapi`

**Comment by pmorie** [comment]
@junxu circling back - do you still need this PR?  If not, can we close it?

**Comment by n3wscott** [comment]
I vote close it @pmorie

File: pkg/brokerapi/service.go (hunk 1 above)
Comment (arschles) [reply pkg/brokerapi/service.go]: +1 on what @duglin said here. I vote for waiting until there's more clarity in the spec

**Comment by nilebox** [comment]
Closing due to inactivity and several Service Catalog approvers voting for closing it.

**Pull request closed without merging**